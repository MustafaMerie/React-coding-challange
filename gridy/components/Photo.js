import React from "react";
import Image from "next/image";
import styles from "../styles/Home.module.css";
import { defaultBlurDataUrl } from "../shared/defaultBlurDataUrl";

function Photo({ photo }) {
  const {
    urls,
    alt_description,
    height,
    width,
    user: { name, profile_image },
  } = photo;

  return (
    <div className={styles.photo_container}>
      <Image
        width={width}
        height={height}
        src={urls.regular}
        alt={alt_description}
        blurDataURL={defaultBlurDataUrl}
        placeholder="blur"
      />
      <div className={styles.photo_on_hover_content}>
        <Image
          width={45}
          height={45}
          src={profile_image.medium}
          alt="user image"
          className="d-inline-block rounded-circle"
        />
        <p className={styles.photo_on_hover_user_name}>{name}</p>
      </div>
    </div>
  );
}

export default Photo;
