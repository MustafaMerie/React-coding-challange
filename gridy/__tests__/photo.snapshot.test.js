import Photo from "../components/Photo";
import renderer from "react-test-renderer";

it("Photo renders correctly", () => {
  const photo = {
    urls: {
      medium:
        "https://images.unsplash.com/photo-1659596357498-70e7c8381b26?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNTAxMDN8MHwxfGFsbHw4fHx8fHx8Mnx8MTY1OTYzMTg3Mg&ixlib=rb-1.2.1&q=80&w=1080",
    },
    alt_description: "THE DESERT\nPhoto by : Pouriya Kafaei",
    height: 6630,
    width: 5304,
    user: {
      name: "pouriya kafaei",
      profile_image: {
        medium:
          "https://images.unsplash.com/profile-1605028582711-88e1234d9466image?ixlib=rb-1.2.1&crop=faces&fit=crop&w=64&h=64",
      },
    },
    links: { html: "https://unsplash.com/@pouriyakafaei" },
  };

  const tree = renderer.create(<Photo photo={photo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
