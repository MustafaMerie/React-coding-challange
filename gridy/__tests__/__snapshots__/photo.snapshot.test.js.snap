// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo renders correctly 1`] = `
<div
  className="photo_container"
  onClick={[Function]}
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%275304%27%20height=%276630%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="THE DESERT
Photo by : Pouriya Kafaei"
      data-nimg="intrinsic"
      decoding="async"
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "backgroundImage": "url(\\"data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAHAAoDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAQH/8QAHhAAAQMFAQEAAAAAAAAAAAAAAQIDEQAEBRIhBpH/xAAUAQEAAAAAAAAAAAAAAAAAAAAD/8QAGBEBAAMBAAAAAAAAAAAAAAAAAQACMRH/2gAMAwEAAhEDEQA/ANacsPasXNw3j2MK8yokt3rjq0qQNhAKCkknWezEkQKnfwnvFPOFu7wGhUSmXbkGJ5zsfaUo2/dCIVTFn//Z\\")",
          "backgroundPosition": "0% 0%",
          "backgroundSize": "cover",
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "filter": "blur(20px)",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="THE DESERT
Photo by : Pouriya Kafaei"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src=""
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  <div
    className="photo_on_hover_content"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2745%27%20height=%2745%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="user image"
        className="d-inline-block rounded-circle"
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "backgroundImage": "url(\\"data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAHAAoDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAQH/8QAHhAAAQMFAQEAAAAAAAAAAAAAAQIDEQAEBRIhBpH/xAAUAQEAAAAAAAAAAAAAAAAAAAAD/8QAGBEBAAMBAAAAAAAAAAAAAAAAAQACMRH/2gAMAwEAAhEDEQA/ANacsPasXNw3j2MK8yokt3rjq0qQNhAKCkknWezEkQKnfwnvFPOFu7wGhUSmXbkGJ5zsfaUo2/dCIVTFn//Z\\")",
            "backgroundPosition": "0% 0%",
            "backgroundSize": "cover",
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "filter": "blur(20px)",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="user image"
          className="d-inline-block rounded-circle"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fprofile-1605028582711-88e1234d9466image%3Fixlib%3Drb-1.2.1%26crop%3Dfaces%26fit%3Dcrop%26w%3D64%26h%3D64&w=96&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fprofile-1605028582711-88e1234d9466image%3Fixlib%3Drb-1.2.1%26crop%3Dfaces%26fit%3Dcrop%26w%3D64%26h%3D64&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fprofile-1605028582711-88e1234d9466image%3Fixlib%3Drb-1.2.1%26crop%3Dfaces%26fit%3Dcrop%26w%3D64%26h%3D64&w=96&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    <p
      className="photo_on_hover_user_name"
    >
      pouriya kafaei
    </p>
  </div>
  <div
    onClick={[Function]}
  />
</div>
`;
